packages/dropdown/src/Dropdown/Dropdown.spec.tsx

```diff
@@ -0,0 +1,229 @@
+      <Dropdown
+        open={open}
+        setOpen={setOpen}
+        triggerRef={triggerRef}
+        className={className}
+        maxWidth={maxWidth}
+        {...props}
+      >
+        <DropdownItem>Dropdown Item A</DropdownItem>
+        <DropdownItem disabled>Dropdown Item B</DropdownItem>
+        <DropdownItem>Dropdown Item C</DropdownItem>
+      </Dropdown>
+    </>
+  );
+};
+
+const WithTriggerExample = (props?: Partial<DropdownProps>) => {
+  const [open, setOpen] = useState(false);
+  const triggerRef = useRef<HTMLButtonElement>(null);
+
+  return (
+    <>
+      <button ref={triggerRef} onClick={() => setOpen(curr => !curr)}>
+        Trigger
+      </button>
+      <Dropdown
+        open={open}
+        setOpen={setOpen}
+        triggerRef={triggerRef}
+        className={className}
+        maxWidth={maxWidth}
+        {...props}
+      >
+        <DropdownItem>Dropdown Item A</DropdownItem>
+        <DropdownItem disabled>Dropdown Item B</DropdownItem>
+        <DropdownItem>Dropdown Item C</DropdownItem>
+        <DropdownItem>Dropdown Item D</DropdownItem>
+      </Dropdown>
+    </>
+  );
+};
+
+const renderDropdown = (props?: Partial<DropdownProps>) => {
+  const utils = render(<DropdownExample {...props} />);
+  return utils;
+};
+
+const renderTriggerExample = (props?: Partial<DropdownProps>) => {
+  const utils = render(<WithTriggerExample {...props} />);
+  return utils;
+};
+
+describe('packages/dropdown', () => {
+  describe('props are applied as expected', () => {
+    test('it renders its children', () => {
+      renderDropdown();
+      const options = screen.getAllByRole('option');
+      expect(options[0]).toBeInTheDocument();
+    });
+
+    test('it adds the className to the class list', () => {
+      renderDropdown();
+      const listbox = screen.getByRole('listbox');
+      expect(listbox.closest(`.${className}`)).toBeInTheDocument();
+    });
+  });
+
+  describe('open and close behavior', () => {
+    test('it renders the listbox when open is true', () => {
+      renderDropdown();
+      const listbox = screen.getByRole('listbox');
+      expect(listbox).toBeInTheDocument();
+    });
+
+    test('it does not render the listbox when open is false', () => {
+      renderDropdown({ open: false });
+      const listbox = screen.queryByRole('listbox');
+      expect(listbox).not.toBeInTheDocument();
+    });
+
+    test('it fires shouldClose when the open state changes inside of the component - backdrop click', () => {
+      renderDropdown({ shouldClose });
+      const backdrop = screen.getByTestId('backdrop');
+      fireEvent.click(backdrop);
+      expect(shouldClose).toHaveBeenCalled();
+    });
+
+    test('it fires the setOpen callback when the open state changes inside of the component - backdrop click', () => {
+      renderDropdown({ setOpen });
+      const backdrop = screen.getByTestId('backdrop');
+      fireEvent.click(backdrop);
+      expect(setOpen).toHaveBeenCalled();
+    });
```

Comment:
should we move these to a mouse interaction section?

Reply:
To me it makes more sense coupled with open/close behavior, but happy to move if you prefer.